internal/component/database_observability/mysql/component.go

```diff
@@ -274,52 +278,20 @@ func (c *Component) Update(args component.Arguments) error {
 
 	c.args = args.(Arguments)
 
-	dbConnection, err := sql.Open("mysql", formatDSN(string(c.args.DataSourceName), "parseTime=true"))
```

Comment:
moving this chunk of code into the startCollectors func.

Reply:
This function doesn't start any collectors, it just opens the SQL connection. The concern I have here is that `Targets` will not have the `server_id` attached as a label through the relabel rules anymore which could cause issues on the query side.

Reply:
Changed it to be a separate metric, called `connection_up`.
Also only exporting the baseTarget when an error is returned by the startCollectors function, which only happens when the connection was unsuccessful or we could not get the engine version.